File: code/222/solution.cpp
#include <iostream>
#include <math.h>
int num[21];
long double cal(int a, int b) {
    long long x = (a + b) * (a + b);
    long long y = (100 - a - b) * (100 - a - b);
    return sqrt(x - y);
}
long double cal() {
    long double ans = num[0] + num[20];
    for (int i = 1; i < 21; i++)
        ans += cal(num[i - 1], num[i]);
    return ans * 1000;
}
int main() {
    // 30 ~ 50
    int i = 0;
    for (int j = 50; j >= 30; j -= 2)
        num[i++] = j;
    for (int j = 31; j <= 50; j += 2)
        num[i++] = j;
    printf("%.0Lf\n", cal());
    return 0;
}

File: code/222/solution.go
package main /*{{{*/

import (
	"fmt"
	"math"
)

var num = [21]float64{}

func cal(a float64, b float64) float64 {
	x := (a + b) * (a + b)
	y := (100 - a - b) * (100 - a - b)
	return math.Sqrt(x - y)
}

func cal2() float64 {
	ans := num[0] + num[20]
	for i := 1; i < 21; i++ {
		ans += cal(num[i-1], num[i])
	}
	return ans * 1000
}

func main() {
	index := 0
	for i := 50; i >= 30; i -= 2 {
		num[index] = float64(i)
		index++
	}
	for i := 31; i <= 50; i += 2 {
		num[index] = float64(i)
		index++
	}
	fmt.Printf("%.0f\n", cal2())
}
